### PR TITLE
[TACHYON-1111] Load Tachyon conf parameters from Hadoop configuration

### PR DESCRIPTION
**Pull request by gpang**
https://tachyon.atlassian.net/browse/TACHYON-1111

### CODE REVIEW

File: clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java [hunk 1]
```diff
@@ -73,9 +74,21 @@ public static TachyonConf loadFromHadoopConfiguration(Configuration source) {
...
+    for (Map.Entry<String, String> entry : source) {
+      String propertyName = entry.getKey();
+      // TODO: a better way to enumerate every Tachyon configuration parameter
+      if (propertyName.startsWith("tachyon.")
+          || propertyName.equals(Constants.S3_ACCESS_KEY)
+          || propertyName.equals(Constants.S3_SECRET_KEY)) {
+        tachyonConfProperties.put(propertyName, entry.getValue());
+      }
+    }
+    LOG.info("Loading Tachyon properties from Hadoop configuration: " + tachyonConfProperties);
```
Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java]: Will this be very noisy?

Comment (apc999) [inline clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java]: probably we should remove it or use `LOG.debug`

File: clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java [hunk 2]
```diff
@@ -73,9 +74,21 @@ public static TachyonConf loadFromHadoopConfiguration(Configuration source) {
...
-
-      return tachyonConfProperties != null ? new TachyonConf(tachyonConfProperties) : null;
     }
-    return null;
+    if (tachyonConfProperties == null) {
+      tachyonConfProperties = new Properties();
+    }
+    // Load any Tachyon configuration parameters existing in the Hadoop configuration.
+    for (Map.Entry<String, String> entry : source) {
+      String propertyName = entry.getKey();
```
Comment (yupeng9) [inline clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java]: Shall we add a unit test for this?

File: clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java (hunk 1 above)
Comment (gpang) [inline clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java]: This is only called by <code>AbstractTFS#initialize</code>. How often is <code>AbstractTFS#initialize</code> called?

File: clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java (hunk 2 above)
Comment (gpang) [inline clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java]: I'm not sure, but I think the tricky part might be that this issue shows up when hadoop command line is remote from the tachyon client, so the unit test probably won't be able to exercise that scenario.

Comment (yupeng9) [inline clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java]: oh, but for unit test it seems we can test the logic by mocking Configuration `source`

File: clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java (hunk 1 above)
Comment (apc999) [inline clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java]: this is called on each client instance. Maybe this is fine so we could see the extracted Hadoop conf per task.

**Comment by apc999** [comment]
+1 LGTM

File: clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java (hunk 2 above)
Comment (gpang) [inline clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java]: I added a simple unit test.

**Comment by yupeng9** [comment]
LGTM

File: clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java (hunk 1 above)
Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java]: Will the output be very large (lots of properties)?

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**